### PR TITLE
Fix #4038 - Re-enable `-Wall` on Unix

### LINKED ISSUE #4038: Re-enable `-Wall` on Unix
Issue overview
--------------

https://github.com/NREL/OpenStudio/commit/8db1b93438e47b234853a2c011061a577458021c had disabled `-Werror` **AND** `-Wall`

https://github.com/NREL/OpenStudio/issues/3577 only re-enabled `-Werror` but not `-Wall`

## Possible Solution

Re-enable `-Wall`, and fix all build errors that will arise in the process.

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version): Ubuntu 18.04
 - Version of OpenStudio (if using an intermediate build, include SHA): develop (post 3.0.1) c68fcf6c0659a31f9538ccfdada1a5d7ea10e431

### Context

`-Wall` is needed. I just got hit with a problem today, where I forgot a `return <object>` in a method that wasn't `void`. `-Wall`, which enables `-Wreturn-type`, would have caught this problem, and that's when I realized it wasn't turned on.

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

- Fix #4038
- Re-enable `-Wall` on Unix

I started doing this on #4037 but I'm having some mac-specific (apple-clang) warnings, so better to separate them in two PRs.

### Pull Request Author

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Checked behavior in OpenStudioApplication, adjusted policies as needed (`src/openstudio_lib/library/OpenStudioPolicy.xml`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: ProjectMacros.cmake [hunk 1]
```diff
@@ -318,7 +318,7 @@ macro(MAKE_SWIG_TARGET NAME SIMPLENAME KEY_I_FILE I_FILES PARENT_TARGET PARENT_S
     if("${CMAKE_CXX_COMPILER_ID}" MATCHES "^(Apple)?Clang$")
       # Prevent excessive warnings from generated swig files, suppress deprecated declarations
       # Suppress 'register' storage class specified warnings (coming from Ruby)
-      set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-Wno-dynamic-class-memaccess -Wno-deprecated-declarations -Wno-sign-compare -Wno-register")
+      set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-Wno-dynamic-class-memaccess -Wno-deprecated-declarations -Wno-sign-compare -Wno-register -Wno-sometimes-uninitialized")
```
Comment (jmarrec) [inline ProjectMacros.cmake:321]:
```suggestion
      set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-Wno-dynamic-class-memaccess -Wno-deprecated-declarations -Wno-sign-compare -Wno-register -Wno-sometimes-uninitialized -Wno-misleading-indentation")
```

File: ProjectMacros.cmake [hunk 2]
```diff
@@ -517,7 +517,7 @@ macro(MAKE_SWIG_TARGET NAME SIMPLENAME KEY_I_FILE I_FILES PARENT_TARGET PARENT_S
       set_target_properties(${swig_target} PROPERTIES SUFFIX ".pyd")
     elseif(UNIX)
       if(APPLE AND NOT CMAKE_COMPILER_IS_GNUCXX)
-        set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-Wno-dynamic-class-memaccess -Wno-deprecated-declarations -Wno-sign-compare")
+        set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-Wno-dynamic-class-memaccess -Wno-deprecated-declarations -Wno-sign-compare -Wno-sometimes-uninitialized")
```
Comment (jmarrec) [inline ProjectMacros.cmake:520]:
```suggestion
        set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-Wno-dynamic-class-memaccess -Wno-deprecated-declarations -Wno-sign-compare -Wno-sometimes-uninitialized -Wno-misleading-indentation")
```

File: ProjectMacros.cmake [hunk 3]
```diff
@@ -756,7 +756,7 @@ macro(MAKE_SWIG_TARGET NAME SIMPLENAME KEY_I_FILE I_FILES PARENT_TARGET PARENT_S
       set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "/bigobj /wd4996") ## /wd4996 suppresses deprecated warnings
       set(final_name "${JAVA_OUTPUT_NAME}.dll")
     elseif(UNIX)
-      set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-Wno-deprecated-declarations -Wno-sign-compare")
+      set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-Wno-deprecated-declarations -Wno-sign-compare -Wno-sometimes-uninitialized")
```
Comment (jmarrec) [inline ProjectMacros.cmake:759]:
```suggestion
      set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-Wno-deprecated-declarations -Wno-sign-compare -Wno-sometimes-uninitialized -Wno-misleading-indentation")
```

File: ProjectMacros.cmake [hunk 4]
```diff
@@ -885,7 +885,7 @@ macro(MAKE_SWIG_TARGET NAME SIMPLENAME KEY_I_FILE I_FILES PARENT_TARGET PARENT_S
     if(MSVC)
       set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "/bigobj /DBUILDING_NODE_EXTENSION /wd4996")  ## /wd4996 suppresses deprecated warnings
     elseif(UNIX)
-      set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-DBUILDING_NODE_EXTENSION -Wno-deprecated-declarations -Wno-sign-compare")
+      set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-DBUILDING_NODE_EXTENSION -Wno-deprecated-declarations -Wno-sign-compare -Wno-sometimes-uninitialized")
```
Comment (jmarrec) [inline ProjectMacros.cmake:888]:
```suggestion
      set_target_properties(${swig_target} PROPERTIES COMPILE_FLAGS "-DBUILDING_NODE_EXTENSION -Wno-deprecated-declarations -Wno-sign-compare -Wno-sometimes-uninitialized -Wno-misleading-indentation")
```

**Comment by tijcolem** [comment]
Looks like some of the compiler options found it's way to the mac builds. 
```
e.g. 
[ 22%] Building CXX object src/measure/CMakeFiles/ruby_OpenStudioMeasure.dir/ruby_OpenStudioMeasure_wrap.cxx.o

error: unknown warning option '-Wno-misleading-indentation'; did you mean '-Wno-binding-in-condition'? [-Werror,-Wunknown-warning-option]

gmake[2]: *** [src/measure/CMakeFiles/ruby_OpenStudioMeasure.dir/build.make:1406: 
```
See https://ci.commercialbuildings.dev/blue/organizations/jenkins/openstudio-incremental-osx/detail/PR-4040/13/pipeline#step-38-log-1285

**Comment by jmarrec** [comment]
Mac and Linux now build correctly. Mac has two tests failing but there are usual suspects (we should fix/disable those):

	614 - BCLFixture.RemoteBCLTest (SIGTRAP)
	616 - BCLFixture.GetComponentByUID (SIGTRAP)

**Pull request merged**